app/Http/Controllers/BLDGApprovalController.php

```diff
@@ -7,13 +7,32 @@
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Validator;
 use App\Models\Service;
+use App\Services\ServicesFromKeyword;
+use Illuminate\Pagination\Paginator;
+use Illuminate\Support\Facades\DB;
 
 class BLDGApprovalController extends Controller
 {
     public function index()
     {
         $services = Service::where('status', 'pending_building_inspection')->paginate(20);
-        return view('pages.bldg-request-approval', ['route' => 'admin.undo', 'text' => ['Lists of Request for Building/Area Inspections', 'View Approved Request'], 'search_heading' => 'SEARCH REQUEST','services' => $services]);
+        return view('pages.bldg-request-approval', ['search_route' => 'admin.search','route' => 'admin.undo', 'text' => ['Lists of Request for Building/Area Inspections', 'View Approved Request'], 'search_heading' => 'SEARCH REQUEST','services' => $services]);
```

Comment:
The data provided on every view is almost the same. I think we can extract a function for this that will build this array for us and just accept parameters based on the data that will change

Reply:
I provide a class which will return array value.

![image](https://user-images.githubusercontent.com/43285621/131917737-c583ef21-c099-48da-bf99-f61ec1f6561f.png)